File: java/insertion.java
import java.util.*;

public class Main {
    public static void main(String[] args) {
      int[] unsorted = {1,5,7,4,10,8,3,2,0,6};
      System.out.print("Unsorted: ");
      for (int i = 0; i < unsorted.length; i++)  {
       System.out.print(unsorted[i] + ",");
     }
      for (int i = 0; i < unsorted.length; i++) {
        int insertion = i;
        while (insertion > 0) {
          if (unsorted[insertion-1] > unsorted[insertion])  {
            int temp = unsorted[insertion];
            unsorted[insertion] = unsorted[insertion-1];
            unsorted[insertion-1] = temp;
          }
          insertion--;
        }
      }
      System.out.print("\nSorted: ");
     for (int i = 0; i < unsorted.length; i++)  {
       System.out.print(unsorted[i] + ",");
     }
  }
}

File: cpp/insertion.cpp
#include <iostream>

using namespace std;

int main()  {
    int unsorted[10] = {7, 2, 9, 5, 1, 8, 3, 10, 6, 4};
    int length = sizeof(unsorted) / sizeof(unsorted[0]);
    for (int i = 0; i < length; i++)    {
        
        for (int j = 1; j < length; j++)    {
            if (unsorted[j] < unsorted[j-1])    {
                int temp = unsorted[j];
                unsorted[j] = unsorted[j-1];
                unsorted[j-1] = temp;
            }
            
        }
    }
    cout << "Sorted array: ";
    for (int i = 0; i < length; i++)    {
        cout << " " << unsorted[i];
    }
    return 0;
}